firebase_admin/_auth_client.py

```diff
@@ -0,0 +1,625 @@
+            provider_id: Provider ID string. Must have the prefix ``saml.``.
+            idp_entity_id: The SAML IdP entity identifier.
+            sso_url: The SAML IdP SSO URL. Must be a valid URL.
+            x509_certificates: The list of SAML IdP X.509 certificates issued by CA for this
+                provider. Multiple certificates are accepted to prevent outages during IdP key
+                rotation (for example ADFS rotates every 10 days). When the Auth server receives a
+                SAML response, it will match the SAML response with the certificate on record.
+                Otherwise the response is rejected. Developers are expected to manage the
+                certificate updates as keys are rotated.
+            rp_entity_id: The SAML relying party (service provider) entity ID. This is defined by
+                the developer but needs to be provided to the SAML IdP.
+            callback_url: Callback URL string. This is fixed and must always be the same as the
+                OAuth redirect URL provisioned by Firebase Auth, unless a custom authDomain is
+                used.
+            display_name: The user-friendly display name to the current configuration (optional).
+                This name is also used as the provider label in the Cloud Console.
+            enabled: A boolean indicating whether the provider configuration is enabled or disabled
+                (optional). A user cannot sign in using a disabled provider.
+
+        Returns:
+            SAMLProviderConfig: The newly created SAMLProviderConfig instance.
+
+        Raises:
+            ValueError: If any of the specified input parameters are invalid.
+            FirebaseError: If an error occurs while creating the new SAML provider config.
+        """
+        return self._provider_manager.create_saml_provider_config(
+            provider_id, idp_entity_id=idp_entity_id, sso_url=sso_url,
+            x509_certificates=x509_certificates, rp_entity_id=rp_entity_id,
+            callback_url=callback_url, display_name=display_name, enabled=enabled)
+
+    def update_saml_provider_config(
+            self, provider_id, idp_entity_id=None, sso_url=None, x509_certificates=None,
+            rp_entity_id=None, callback_url=None, display_name=None, enabled=None):
+        """Updates an existing SAML provider config with the given parameters.
+
+        Args:
+            provider_id: Provider ID string. Must have the prefix ``saml.``.
+            idp_entity_id: The SAML IdP entity identifier (optional).
+            sso_url: The SAML IdP SSO URL. Must be a valid URL (optional).
+            x509_certificates: The list of SAML IdP X.509 certificates issued by CA for this
+                provider  (optional).
+            rp_entity_id: The SAML relying party entity ID (optional).
+            callback_url: Callback URL string  (optional).
+            display_name: The user-friendly display name to the current configuration (optional).
```

Comment:
Unless this is technically required, "name of" sounds more natural.

Reply:
Done